core/spec/models/spree/order_merger_spec.rb

```diff
@@ -141,20 +141,6 @@ module Spree
         expect(order_1.line_items.pluck(:quantity)).to match_array([1, 1])
         expect(order_1.line_items.pluck(:variant_id)).to match_array([variant.id, variant_2.id])
       end
-
-      context "with line item promotion applied to order 2" do
-        let!(:promotion) { create(:promotion, :with_line_item_adjustment, apply_automatically: true) }
-
-        before do
-          Spree::PromotionHandler::Cart.new(order_2).activate
```

Comment:
Not sure about moving this spec. I feel like the `Spree::PromotionHandler::Cart` here has been used only to generate a line item adjustment. This test verifies that we don't bring wrong line item adjustments over the merging object, and might be more related to how the merger works than to this specific promotional line item adjustment.

Reply:
Right. I've changed the commit so that we copy the original spec and modify the one in core to use a manual line item adjustment instead.

Reply:
Thanks! This spec should have been done this way since the beginning.